_posts/2020-04-03-why-did-we-switch-to-quarkus.adoc

```diff
@@ -0,0 +1,85 @@
+
+* *Reduction in reflection usage.* Considering the point above, the resulting application contains only direct invocations. All of the fields, methods and classes that are not being called directly are purposefully removed. The same applies to dependencies as well. There is a way to include such calls (register them for reflection), but Quarkus will gently warn you that you may be deflecting from the real purpose of the framework.
+
+* *Native image pre-boot.* Quarkus supports native executables that can be run with GraalVM. Actually, creators of Quarkus state that the framework was designed to be run as a native image. The ahead-of-time compilation reduces the need of dynamic scanning and loading classes into a JVM.
+
+* *GraalVM.* GraalVM is a native code generator which will compile your project into an executable that can be run without the JVM. You can read more about this superb project here: https://www.graalvm.org/.
+
+Below you can see a comparison of memory usage and boot time of sample applications:
+
+image::perf_black.png[Performance Chart]
+source: https://quarkus.io/
+
+== Evolution, not revolution
+
+What is interesting about this framework is that it is not revolutionary. Since some of Payair’s developers had mainly Spring experience, we were concerned that it would be difficult for them to “switch sides”. It turned out that our fear of the unknown was completely unfounded. Quarkus is built on the foundations of good old Jakarta EE standards that all Java developers are familiar with. We did not have to learn a bunch of new APIs that are advertised as "the best way to create REST endpoint" or "reinvented database access". Instead, Quarkus uses Context Dependency Injection for beans, JAX-RS for RESTful web services, JPA for database access and many others.
+
+If you happen to write your project using Jakarta EE, switching to Quarkus would be as easy as changing the runtime processor. Majority of your old code would work without any adjustments.
+
+On top of that, Quarkus extends available Jakarta EE functionalities responding to modern world developers' demands. These are called Quarkus extensions and, to name a few, they contain support for reactive programming (but using the declarative approach), event messaging, etc.
+
+Apart from a wide variety of available extensions, there is also a way to create your own, based on the library of your choice. Keep in mind though, that if you want to preserve those drastic optimizations that Quarkus offers, you will have to apply them manually to your extension as well.
+
+== Shout-out to MicroProfile
+
+I would like to stop here for a second and give a big shout-out to Eclipse MicroProfile initiative. It specifies a collection of Jakarta EE APIs and technologies to form a baseline for quick and effective microservices creation. The goal of MicroProfile is to define a standard for building microservices and deliver portable applications across multiple Jakarta EE runtime vendors.
```

Comment:
Suggest the following slight modification:
"I would like to stop here for a second and give a big shout-out to Eclipse MicroProfile initiative. It specifies a collection of APIs and technologies to form a baseline for quick and effective enterprise Java microservices creation. The goal of MicroProfile is to define a standard for building microservices and deliver portable applications across multiple MicroProfile runtime implementations."